### ISSUE TITLE
Property changes in DidInsertElement

### ISSUE THREAD

**Issue by ianbale** [AddJam/ember-content-editable#11]
Hi,

I've spotted another issue. Using this with attributes in my model, it works fine. However, my model has a hasMany relationship with some other data. Attempting to use it with that results in this error:

ember.debug.js:5523 DEPRECATION: A property of <ember-accounts@view:-outlet::ember499> was modified inside the didInsertElement hook. You should never change properties on components, services or models during didInsertElement because it causes significant performance degradation.
        at _emberViewsViewsCore_view.default.extend.scheduleRevalidate (http://localhost:4200/assets/vendor.js:61168:34)

I've no idea why it should be handling this data differently? I've tried using {{input}} instead and that seems to work, so I'm assuming that the issue resides in this control, not in my code. I stand to be corrected though! 

Hopefully, what I have is pretty easy to replicate... Here's my data ( I get this via a JSON API, but I'm assuming we'll have the same issue if it's done as a fixture):

{
  "data": {
    "id": 1,
    "attributes": {
      "created": "November, 01 2000 00:00:00",
      "number": "A0011000001"
    },
    "type": "invoice",
    "relationships": {
      "details": {
        "data": [
          {
            "id": 1,
            "type": "invoice-detail"
          }
        ]
      },
      "client": {
        "data": {
          "id": 14,
          "type": "client"
        }
      },
      "user": {
        "data": {
          "id": 2,
          "type": "user"
        }
      }
    }
  },
  "included": [
    {
      "id": 1,
      "attributes": {
        "amount": 3000,
        "detail": "Stage 1 delivery of 3Com Reseller Locator to\r\nFoundation Network LTD",
        "vat": 525
      },
      "type": "invoice-detail"
    }
  ]
}

And here's some simplified code:

{{content-editable value=model.number isText=true}}

{{#each model.details as |detai|}}
	{{content-editable value=detail.amount isText=true}}
{{/each}}

The first content-editable works fine. Its the one accessing the model.details (which is in the "included" part of the data) that results in the warning.

I'll take a look myself at some point and let you know if I spot something.

**Comment by chrissloey** [comment]
I was able to reproduce using the value `Stage 1 delivery of 3Com Reseller Locator to\r\nFoundation Network LTD`.  I just released a fix in v0.3.10 :smile:

This was caused because in didInsertElement the component first sets the provided value on the element in the DOM, and then processes the value in the element as if it was entered by the user and sets it back on the bound value. I think this is only coming up now because of the `\r\n` characters. Anyway, I've delayed the part that updates the bound value.
